From ef1f0fc033c453f6652190527ab0a0f228b16335 Mon Sep 17 00:00:00 2001
From: Baptiste Assmann <bedis9@gmail.com>
Date: Sun, 23 Aug 2015 10:06:39 +0200
Subject: [PATCH 12/28] DOC: new global directive: server-state-file

Documentation related to a new global directive.
Purpose of this directive is to store a file path into the global
structure of HAProxy. The file pointed by the path may be used by
HAProxy to retrieve server state from the previous running process
after a reload occured.
---
 doc/configuration.txt | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/doc/configuration.txt b/doc/configuration.txt
index 7508a28..4d6676f 100644
--- a/doc/configuration.txt
+++ b/doc/configuration.txt
@@ -562,6 +562,7 @@ The following keywords are supported in the "global" section :
    - nogetaddrinfo
    - spread-checks
    - server-state-base
+   - server-state-file
    - tune.bufsize
    - tune.chksize
    - tune.comp.maxlevel
@@ -762,7 +763,18 @@ stats bind-process [ all | odd | even | <number 1-64>[-<number 1-64>] ] ...
 
 server-state-base <directory>
   Specifies the directory prefix to be prepended in front of all servers state
-  file names which do not start with a '/'. Read below 'See also'.
+  file names which do not start with a '/'. See also "server-state-file".
+
+server-state-file <file>
+  Specifies the path to the file containing state of servers. If the path starts
+  with a slash ('/'), it is considered absolute, otherwise it is considered
+  relative to the directory specified using "server-state-base" (if set) or to
+  the current directory. Before reloading HAProxy, it is possible to save the
+  servers' current state using the stats command "show servers state". The
+  output of this command must be written in the file pointed by <file>. When
+  starting up, before handling traffic, HAProxy will read, load and apply state
+  for each server found in the file and available in its current running
+  configuration. See also "server-state-base" and "show servers state".
 
 ssl-default-bind-ciphers <ciphers>
   This setting is only available when support for OpenSSL was built in. It sets
-- 
2.4.9

